R/model.R

```diff
@@ -1360,33 +1395,196 @@ artnum_mf <- function(calendar_quarter, art_number, naomi_mf) {
+      dplyr::select(all_of(cols))
+    #  Tag row selected as model inputs
+    full_artnum$calendar_quarter <- calendar_quarter
+    full_artnum$data_type <- "interpolated"
     }
 
-    if (nrow(dat) == 0) {
+    if (nrow(full_artnum) == 0) {
       stop(t_("NO_ART_DATA_FOR_QUARTER",
               list(calendar_quarter = calendar_quarter)))
     }
 
-    artnum_dat <- dat %>%
-      dplyr::transmute(
-               area_id,
-               sex,
-               age_group,
-               artnum_idx = dplyr::row_number(),
-               art_current
-             )
+
+    ## Calculate model inputs for all data provided
+    ## Filter observations contained in model scope
+    filtered_artnum <- dplyr::semi_join(
+      full_artnum,
+      naomi_mf$mf_model,
+      by = "area_id"
+    ) %>%
+      dplyr::mutate(data_type = paste0(data_type, "_included"))
+
+    #  Get intersecting colnames to silence anti_join message
+    col_names <- intersect(names(full_artnum), names(filtered_artnum))
+
+    excluded <- dplyr::anti_join(full_artnum, filtered_artnum, by = col_names) %>%
+      dplyr::mutate(data_type = paste0(data_type, "_excluded"))
+
+
+    full_artnum <- rbind(filtered_artnum, excluded)
+    filtered_artnum <- dplyr::select(filtered_artnum,
+                                     area_id, sex, age_group, art_current)
+
+    artnum_out <- list(raw_input = full_artnum,
+                      model_input = filtered_artnum)
+
   }
 
-  artnum_dat
 }
+
+
+
+#' @param naomi_data Naomi object of class "naomi_data" and "naomi_mf"
+#'
+#' @details
+#'
+#'
+#' @export
+
+
+extract_naomi_inputs <- function(naomi_data){
+
+  stopifnot(inherits(naomi_data, "naomi_data"))
+  stopifnot(inherits(naomi_data, "naomi_mf"))
+
+  # Collate and format data inputs to match outputs with column names:
+  #  "area_id", "sex" , "age_group" , "calendar_quarter", "indicator","mean",
+  # "se" ,"median" , "mode" , "lower" , "upper", "data_type"
+
+  format_inputs_to_outputs <- function(dat, source = "programme_data") {
+
+    indicators <- c("art_current", "art_new", "vl_tested_12mos", "vl_supressed_12mos",
+                    "anc_clients", "anc_known_pos","anc_already_art", "anc_tested",
+                    "anc_tested_pos")
+
+
+    ind <- intersect(colnames(dat), indicators)
+
+    long <- dat %>%
+      dplyr::select(area_id,  sex , age_group, calendar_quarter, data_type, all_of(ind)) %>%
+      tidyr::pivot_longer(cols = all_of(ind),
+                          names_to = "indicator", values_to = "mean") %>%
+      dplyr::mutate(se = NA_real_ ,median = NA_real_ , mode = NA_real_ ,
+                    lower = NA_real_ , upper = NA_real_, source = source)
+
+
+  }
+
+  # Collate and format survey_data
+  input_dat <- rbind(naomi_data$prev_dat$raw_input,
+                     naomi_data$artcov_dat$raw_input,
+                     naomi_data$recent_dat$raw_input,
+                     naomi_data$vls_dat$raw_input) %>%
```

Comment:
I can't find where in the code this is getting assigned; but leaving comment here that we discussed.

 Survey data input issue: survey could be selected but survey_mid_calendar_quarter
 is a different quarter than selected quarter. -> cannot use survey_mid_calendar_quarter
 to join to outputs

 Resolution: 
 * Retain column survey_mid_calendar_quarter unchanged
 * Assign new column `calendar_quarter` that takes calendar quarter T1 if
   if the survey is selected, is NA otherwise.